File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-210220-97414/analysis/acceptance-project/src/surveycrawler/advance_command.py
from surveycrawler.command import Command


class AdvanceCommand(Command):

    # character

    @classmethod
    def character(cls):
        return 'a'

    # executing

    def execute_on(self, a_crawler):
        a_crawler.advance()

    def undo_on(self, a_crawler):
        a_crawler.move_one_cell_backward()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-204934-85557/output/project/src/main/java/surveycrawler/AdvanceCommand.java
package surveycrawler;

public class AdvanceCommand extends CrawlerMovement {

    // testing

    @Override
    public boolean isFor(char aCharacter) {
        return aCharacter == 'a';
    }

    // executing

    @Override
    public void executeOn(SurveyCrawler aCrawler) {
        aCrawler.advance();
    }

    // undoing

    @Override
    public CrawlerMovement inverse() {
        return new RetreatCommand();
    }
}
